### PR TITLE
Support Non Top Level Scope for Bridged Classes Built From Functions

### PR DESCRIPTION
**Pull request by wied03**
This could have bigger implications then what I realize but rather than just complain (see issue https://github.com/opal/opal/issues/883), I thought I'd submit a PR.

### CODE REVIEW

**Comment by wied03** [comment]
This issue was noticed when trying to use opal-jquery and react.rb in the same project and namespace collisions from the 'Element' class quickly surfaced.

**Comment by elia** [comment]
Thanks!

I'll add just a couple of comments on styling, let me know if you can update the PR, otherwise I'll merge and then apply the fixes myself.

File: opal/corelib/runtime.js [hunk 1]
```diff
@@ -560,14 +560,22 @@
...
    */
-  function bridge_class(name, constructor) {
+  function bridge_class(name, constructor, base) {
     var klass = boot_class_object(ObjectClass, constructor);
 
     klass.$$name = name;
 
-    create_scope(Opal, klass, name);
+	if (base === undefined) {
+		base = Opal;
```
Comment (elia) [inline opal/corelib/runtime.js]: Indentation should be 2 spaces here

File: spec/opal/core/runtime/bridged_classes_spec.rb [hunk 2]
```diff
@@ -62,3 +61,51 @@ def some_bridged_method
...
+  it "instances of class should be able to call native ruby methods" do
+    @instance.foo.should == "foobar"
+    @bridged.new.foo.should == "foobar"
+  end
+
+  it "allows new methods to be defined on the bridged prototype" do
+    @instance.some_bridged_method.should == [4, 5, 6]
+    @bridged.new.some_bridged_method.should == [4, 5, 6]
+  end
+end
```
Comment (elia) [inline spec/opal/core/runtime/bridged_classes_spec.rb]: missing EOF newline

**Comment by elia** [comment]
that's what I meant thanks!

**Pull request merged**